silenttrinity/core/teamserver/contexts/sessions.py

```diff
@@ -235,10 +239,42 @@ def checkin(self, guid):
     def rename(self, guid, name):
         try:
             session = self.get_session(guid)
-            session.guid = name
+            session.name = name
         except SessionNotFoundError:
             raise CmdError(f"No session with guid: {guid}")
 
+    def unregister(self, guid):
+        self.guidchk(guid)
+        with STDatabase() as db:
+            psk = db.get_session_psk(guid)
+        session = Session(guid, psk)
+        self.sessions.add(session)
+
+        if (session in self.sessions):
+            return {"message": "You can't kill an active session. Kill and purge first."}
+        with STDatabase() as db:
+            db.remove_session(guid)
```

Comment:
Perhaps hiding the session from UI instead of removing is a safer approach in case there are temporary connectivity issues (infected machine went to sleep) so the session might reconnect back later? Otherwise (correct me if I'm wrong) the stager will just keep banging the door, but won't get re-registered due to lack of matching guid/psk in db?

Reply:
That is correct.

I made the purge command for that. If you unregister the session any session using that guid/psk is “dead” unless you add it back with the register command.

I did want a way to remove it from the DB if needed as I was also testing some stagers that continues to add sessions to the DB and I wanted a way to clean them up from the without manually modifying the DB.

Should I rename it? Add better docs? Trash it?

It may be useless to others.

Reply:
This is awesome! Purge to unclutter the UI, remove to invalidate session completely :+1: